### PR TITLE
EoU u30 CRs

### PR DESCRIPTION
**Pull request by AShivangi**
### CRs addressed:
- CR-1091797 xbutil examine CU status gives a false negative
- CR-1091660 Xbutil examine shows bug that xbutil query does not
- CR-1091505 Modify verbiage to keep cards vs devices straight during programming
- CR-1091489 xbmgmt examine -r host is reporting in user space
- CR-1091093 xbmgmt2 programming command error references an undocumented --update switch
- CR-1091091 Please have programming operations work for all u30 devices

### CODE REVIEW

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 1]
```diff
@@ -70,9 +74,28 @@ get_cu_status(uint32_t cu_status)
   return pt;
 }
 
+static void
+schedulerUpdateStat(xrt_core::device *device)
+{
+  try {
+    // lock xclbin
+    auto uuid = xrt::uuid(xrt_core::device_query<xrt_core::query::xclbin_uuid>(device));
+    device->open_context(uuid.get(), static_cast<unsigned int>(-1), true);
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportCu.cpp:83]: Please don't use -1 to initialize an unsigned int to all 1s.   Instead please use:
`std::numeric_limits<unsigned int>::max()`

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 2]
```diff
@@ -70,9 +74,28 @@ get_cu_status(uint32_t cu_status)
...
 }
 
+static void
+schedulerUpdateStat(xrt_core::device *device)
+{
+  try {
+    // lock xclbin
+    auto uuid = xrt::uuid(xrt_core::device_query<xrt_core::query::xclbin_uuid>(device));
+    device->open_context(uuid.get(), static_cast<unsigned int>(-1), true);
+    auto at_exit = [] (auto device, auto uuid) { device->close_context(uuid.get(), static_cast<unsigned int>(-1)); };
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportCu.cpp:84]: Please use: `std::numeric_limits<unsigned int>::max()`

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 3]
```diff
@@ -70,9 +74,28 @@ get_cu_status(uint32_t cu_status)
...
+  }
+  catch (const std::exception&) {
+    // xclbin_lock failed, safe to ignore
+  }
+}
+
 boost::property_tree::ptree
 populate_cus(const xrt_core::device *device)
 {
+  if (!std::getenv("XCL_SKIP_CU_READ"))
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportCu.cpp:97]: Could you help me understand why we are looking at environment variables.  Unfortunately, this is a bad practice and should be avoided whenever possible.

File: src/runtime_src/tools/xclbinutil/CMakeLists.txt [hunk 4]
```diff
@@ -9,7 +9,7 @@ endif()
 # -----------------------------------------------------------------------------
 # Include the generated header files (e.g., version.h)
 include_directories(${CMAKE_BINARY_DIR}/gen)
-
+RETURN()
```
Comment (rozumx) [inline src/runtime_src/tools/xclbinutil/CMakeLists.txt:12]: Hmm...did you mean to do this?

**Review by rozumx — CHANGES_REQUESTED** [review]
Some minor changes requested.

Double Check Request:  Since we are rebranding the "Shell" to "Base" and "Card" to "Device", please do a string search to see if you got all of the places they are used :-).

File: src/runtime_src/tools/xclbinutil/CMakeLists.txt (hunk 4 above)
Comment (AShivangi) [reply src/runtime_src/tools/xclbinutil/CMakeLists.txt]: Nope!

File: src/runtime_src/core/tools/common/ReportCu.cpp (hunk 2 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/common/ReportCu.cpp]: fixed

File: src/runtime_src/core/tools/common/ReportCu.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/common/ReportCu.cpp]: fixed

File: src/runtime_src/core/tools/common/ReportCu.cpp (hunk 3 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/common/ReportCu.cpp]: Not needed. removed

**Review by rozumx — APPROVED** [review]
Thank you for making these changes.

**Pull request merged**